jwst/residual_fringe/utils.py

```diff
@@ -840,3 +841,408 @@ def new_make_knots(flux, nknots=20, weights=None):
+        try:
+            pars = ftr.fit(res_fringes, weights=weights)
+
+            # free the parameters and refit
+            mdl = multi_sine(nfringes + 1)
+            mdl.parameters = pars
+            fitter = LevenbergMarquardtFitter(wavenum, mdl, verbose=0)
+            ftr = RobustShell(fitter, domain=10)
+            pars = ftr.fit(res_fringes, weights=weights)
+
+            # try get evidence (may fail for large component fits to noisy data, set to very negative value
+            try:
+                evidence2 = fitter.getEvidence(limits=[-2, 1000], noiseLimits=[0.001, 1])
+            except ValueError:
+                evidence2 = -1e9
+        except RuntimeError:
+            evidence2 = -1e9
+
+        # log.debug("fit_1d_fringes_bayes_evidence: nfringe={} ev={} chi={}".format(nfringes, evidence2, fitter.chisq))
+
+        bayes_factor = evidence2 - evidence1
+        # log.debug(
+        #    "fit_1d_fringes_bayes_evidence: bayes factor={}".format(bayes_factor))
+        if bayes_factor > 1:  # strong evidence thresh (log(bayes factor)>1, Kass and Raftery 1995)
+            evidence1 = evidence2
+            best_mdl = mdl.copy()
+            fitted_frequencies.append(freqs)
+            # log.debug(
+            #    "fit_1d_fringes_bayes_evidence: strong evidence for nfringes={} ".format(nfringes + 1))
+        else:
+            # log.debug(
+            #    "fit_1d_fringes_bayes_evidence: no evidence for nfringes={}".format(nfringes + 1))
+            break
+
+        # subtract the fringes for this frequency
+        res_fringe_fit = best_mdl(wavenum)
+        res_fringes_proc = res_fringes.copy() - res_fringe_fit
+        nfringes += 1
+
+    if plots:
+        if ffreq < 1:
+            axs[0].set_title(r'Fringe #2: freq-by-freq removal and periodograms')
+        else:
+            axs[0].set_title(r'Fringe #1: freq-by-freq removal and periodograms')
+        axs[0].set_xlabel(r'Wavenumber (cm$^{-1})$')
+        axs[0].set_ylabel(r'Residual fringes')
+        axs[0].set_ylim(-0.2, (n+1)*0.12)
+        axs[0].grid()
+        axs[1].set_xlabel(r'Fringe frequency (cm$^{-1}$)')
+        axs[1].set_ylabel(r'Power')
+        axs[1].grid()
+        axs[0].legend()
+        axs[1].legend()
+        plt.tight_layout(h_pad=0)
+        plt.show()
+
+        # log.debug("fit_1d_fringes_bayes_evidence: optimal={} fringes".format(nfringes))
+        print(" optimal={} fringes".format(nfringes))
+
+    # create outputs to return
+    fitted_frequencies = (1 / np.asarray(fitted_frequencies)) * factor
+    peak_freq = fitted_frequencies[0]
+    freq_min = np.amin(fitted_frequencies)
+    freq_max = np.amax(fitted_frequencies)
+
+    return res_fringe_fit, weighted_pix_num, nfringes, peak_freq, freq_min, freq_max
+
+# function to run the fitting on an x1d file and save the output
+def fit_residual_fringes_1d(flux, wavelength, channel=1, save_results=True, dichroic_only=False, max_amp=None):
```

Comment:
`save_results` does not seem to be used anywhere or did I miss it?

Reply:
I don't think it is- deleted.